packages/amplify_authenticator/lib/src/widgets/form.dart

```diff
@@ -262,47 +232,125 @@ class SignUpForm extends AuthenticatorForm {
   /// {@macro amplify_authenticator.sign_up_form}
   SignUpForm({
     Key? key,
-  }) : this.custom(
-          fields: [],
+  })  : includeSignUpAttributes = true,
+        includeVerificationAttributes = true,
+        super._(
           key: key,
+          fields: requiredFields,
+          actions: const [
+            SignUpButton(),
+          ],
         );
 
+  /// The minimum required attributes for sign up.
+  ///
+  /// This does not include sign up attributes or verification attributes
+  /// that may be required by cognito based on your cognito configuration.
+  /// See [includeSignUpAttributes] and [includeVerificationAttributes].
+  static List<SignUpFormField> requiredFields = [
+    SignUpFormField.username(),
+    SignUpFormField.password(),
+    SignUpFormField.passwordConfirmation(),
+  ];
+
   /// A custom Sign Up form.
-  SignUpForm.custom({
+  const SignUpForm.custom({
     Key? key,
+    this.includeSignUpAttributes = false,
+    this.includeVerificationAttributes = false,
     required List<SignUpFormField> fields,
   }) : super._(
           key: key,
-          requiredFields: [
-            SignUpFormField.username(),
-            SignUpFormField.password(),
-            SignUpFormField.passwordConfirmation(),
-          ],
-          customFields: fields,
+          fields: fields,
           actions: const [
             SignUpButton(),
           ],
         );
 
+  /// Determines if sign up attributes from the auth config will
+  /// automatically be added to the form at runtime
+  ///
+  /// If custom attributes are provided that overlap with any of the sign up
+  /// attributes from the config, the custom attributes will be used instead.
+  ///
+  /// If set to true, all displayed attributes will automatically be sorted
+  final bool includeSignUpAttributes;
+
+  /// Determines if verification attributes from the auth config will
+  /// automatically be added to the form at runtime
+  ///
+  /// If custom attributes are provided that overlap with any of the sign up
+  /// attributes from the config, the custom attributes will be used instead.
+  ///
+  /// If set to true, all displayed attributes will automatically be sorted
+  final bool includeVerificationAttributes;
+
   @override
   _SignUpFormState createState() => _SignUpFormState();
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty<bool>(
+      'includeSignUpAttributes',
+      includeSignUpAttributes,
+    ));
+    properties.add(DiagnosticsProperty<bool>(
+      'includeVerificationAttributes',
+      includeVerificationAttributes,
+    ));
+  }
 }
 
 class _SignUpFormState extends AuthenticatorFormState<SignUpForm> {
   _SignUpFormState() : super();
 
   @override
-  List<SignUpFormField> runtimeFields(BuildContext context) {
+  List<SignUpFormField> get allFields {
```

Comment:
Should this logic be located here? Seems like it might be needed in other forms as well.

Reply:
It isn't used in any other forms currently. It's possible that it could be used in other forms in the future, but I cannot think of a use case for it.